src/Resolver/ResourceToTeaserResolver.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Atoolo\GraphQL\Search\Resolver;
+
+use Atoolo\GraphQL\Search\Types\ArticleTeaser;
+use Atoolo\GraphQL\Search\Types\MediaTeaser;
+use Atoolo\GraphQL\Search\Types\Teaser;
+use Atoolo\Resource\Resource;
+use Overblog\GraphQLBundle\Definition\ArgumentInterface;
+
+class ResourceToTeaserResolver
```

Comment:
I'dont think the name is all to descriptive here as both, `ResourceToTeaserResolver` and `TeaserResolver` resolve `Resource`s to `Teaser`.  
The name `Resolver` is used up for classes that can be registered in the `ResolverMapRegistry` (if I understand this correctly) so this class should be named `TeaserResolver` and the interface currently named that shouldn't.   
How about we name that `TeaserFactory` instead?

Reply:
Yes, the fact that the term 'resolver' is used for two different cases is very confusing. I have now changed it so that 'Resolver' is really only used for the 'ResolverMapRegistry'.

The teasers are now created via a teaser factory. `ResourceToTeaserResolver => TeaserResolver` is the `DelegatingTeaserFactory` which itself also implements the `TeaserFactory` interface. This makes it even easier to replace.